execution_engine/src/core/engine_state/mod.rs

```diff
@@ -995,11 +993,9 @@ where
 
                 let contract_hash = *contract_package
                     .lookup_contract_hash(contract_version_key)
-                    .ok_or_else(|| {
-                        error::Error::Exec(execution::Error::InvalidContractVersion(
-                            contract_version_key,
-                        ))
-                    })?;
+                    .ok_or(error::Error::Exec(
+                        execution::Error::InvalidContractVersion(contract_version_key),
+                    ))?;
```

Comment:
How do these work now? Will it construct an instead of the error lazily or eagerly?

Reply:
[They're constructed eagerly](https://doc.rust-lang.org/std/option/enum.Option.html#method.ok_or).  I guess since it's just a simple constructor in there rather than a function proper, [the lint](https://rust-lang.github.io/rust-clippy/master/index.html#unnecessary_lazy_evaluations) has changed.